src/test/java/org/jenkinsci/plugins/Jenkins47113.java

```diff
@@ -0,0 +1,315 @@
+                    break;
+                case "/login/oauth/authorize":
+                    this.onLoginOAuthAuthorize(req, resp);
+                    break;
+                case "/login/oauth/access_token":
+                    this.onLoginOAuthAccessToken(req, resp);
+                    break;
+            }
+        }
+
+        private void onUser(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+            resp.getWriter().write(JSONObject.fromObject(
+                    new HashMap<String, Object>() {{
+                        put("login", currentLogin);
+                        put("name", currentLogin + "_name");
+                        // to avoid triggering a second call, due to GithubSecurityRealm:382
+                        put("created_at", "2008-01-14T04:33:35Z");
+                        put("url", serverUri + "/users/_specific_login_");
+                    }}
+            ).toString());
+        }
+
+        private void onUserOrgs(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+            List<Map<String, Object>> responseBody = new ArrayList<>();
+            for (String orgName : organizations) {
+                final String orgName_ = orgName;
+                responseBody.add(new HashMap<String, Object>() {{
+                    put("login", orgName_);
+                }});
+            }
+
+            resp.getWriter().write(JSONArray.fromObject(responseBody).toString());
+        }
+
+        private void onUserTeams(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+            List<Map<String, Object>> responseBody = new ArrayList<>();
+            for (String teamName : teams) {
+                final String teamName_ = teamName;
+                responseBody.add(new HashMap<String, Object>() {{
+                    put("login", teamName_ + "_login");
+                    put("name", teamName_);
+                    put("organization", new HashMap<String, Object>() {{
+                        put("login", organizations.get(0));
+                    }});
+                }});
+            }
+
+            resp.getWriter().write(JSONArray.fromObject(responseBody).toString());
+        }
+
+        private void onLoginOAuthAuthorize(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+            String code = "test";
+            resp.sendRedirect(staticJenkins.getURL() + "securityRealm/finishLogin?code=" + code);
+        }
+
+        private void onLoginOAuthAccessToken(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+            resp.getWriter().write("access_token=RANDOM_ACCESS_TOKEN");
+        }
+    }
+
+    @AfterClass
+    public static void stopJetty() {
+        try {
+            server.stop();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Issue("JENKINS-47113")
+    @Test
+    public void testGroupPopulation() throws Exception {
```

Comment:
I'd consider moving this stuff to an `@Before` method and adding `@Test` to `testAlice_usingGithubToken` and `testBob_usingGithubLogin`, and maybe renaming those methods to `testLoginUsingGithubApiToken` and `testLoginUsingGithubOauth`?

Reply:
For that point I followed some previous comments from Jesse, telling me it's better to have a single test in order to avoid double Jenkins startup.
But in this plugin, there's not a lot of tests, so you're right I think, will do the change 👍